app/pathogen/arbovirus/dashboard/(visualizations)/median-seroprevalence-by-who-region-and-age-group-table.tsx

```diff
@@ -0,0 +1,191 @@
+import { useContext, useMemo, useState } from "react";
+
+import { ArboContext } from "@/contexts/arbo-context/arbo-context";
+import {
+  mixColours,
+  typedGroupBy,
+  typedObjectEntries,
+  typedObjectFromEntries,
+  typedObjectKeys,
+} from "@/lib/utils";
+import { arbovirusesSF, convertArboSFtoArbo, median } from "./recharts";
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from "@/components/ui/table";
+import { WHORegion } from "@/lib/who-regions";
+import { Button } from "@/components/ui/button";
+
+enum AgeGroup {
+  "Adults (18-64 years)" = "Adults (18-64 years)",
+  "Children and Youth (0-17 years)" = "Children and Youth (0-17 years)",
+  "Seniors (65+ years)" = "Seniors (65+ years)",
+  "Multiple groups" = "Multiple groups",
+}
+
+export const MedianSeroprevalenceByWhoRegionAndAgeGroupTable = () => {
+  const state = useContext(ArboContext);
+  const pathogenOrder: arbovirusesSF[] = [
+    "ZIKV",
+    "DENV",
+    "CHIKV",
+    "YF",
+    "WNV",
+    "MAYV",
+  ];
+  const [currentIndex, setCurrentIndex] = useState(0);
+
+  const datasetGroupedByArbovirus = useMemo(
```

Comment:
Yeah I don't really like what I'm doing with the grouping functions either. I want to make a more generalized triple group function like we have with the bar charts double grouping function but figured it doesn't really have to happen right now right now.

`useMemo`s for performance ofc